DataFormats/TestObjects/interface/ThingWithDoNotSort.h

```diff
@@ -0,0 +1,30 @@
+#ifndef DataFormats_TestObjects_interface_ThingWithDoNotSort_h
+#define DataFormats_TestObjects_interface_ThingWithDoNotSort_h
+
+#include <stdexcept>
+#include <cstdint>
+
+#include "DataFormats/Common/interface/traits.h"
+#include "FWCore/Utilities/interface/typedefs.h"
+
+namespace edmtest {
+
+  class ThingWithDoNotSort : public edm::DoNotSortUponInsertion {
+  public:
+    ThingWithDoNotSort() : value_{0} {};
+    explicit ThingWithDoNotSort(cms_int32_t v) : value_{v} {}
+
+    void post_insert() {
+      throw std::logic_error(
```

Comment:
Ideally this would be `cms::Exception` (sorry, it caught my eye only now). However, I see the only exception presently being thrown in `DataFormats/TestObjects` is also `std::logic_error` (seems like a remnant from 2006), so we could proceed with `logic_error` now and change them both to `cms::Exception` in a subsequent PR.

Reply:
Done.